src/reader/bytes.rs

```diff
@@ -0,0 +1,906 @@
+//! A module to handle `XmlBytesReader`
+
+use std::fmt;
+use std::fs::File;
+use std::io::{self, BufRead, BufReader};
+use std::path::Path;
+use std::borrow::Cow;
+use std::ops::Range;
+
+use error::{Error, Result, ResultPos};
+use escape::unescape;
+use reader::attributes::{Attributes, UnescapedAttributes};
+use AsStr;
+
+#[derive(Clone)]
+enum TagState {
+    Opened,
+    Closed,
+//     Empty,
+}
+
+/// General content of an event (aka node)
+///
+/// BytesElement is a wrapper over the bytes representing the node:
+///
+/// E.g. given a node `<name att1="a", att2="b">`, the corresponding `BytesEvent` will be
+///
+/// ```ignore
+/// BytesEvent::Start(BytesElement {
+///     buf:    b"name att1=\"a\", att2=\"b\"",
+///     start:  0,
```

Comment:
This is out of sync with the code which only members `buf`, `content`, and `name`.